ledger/src/blockstore.rs

```diff
@@ -2872,20 +2872,24 @@ impl Blockstore {
                     .and_then(|serialized_shred| {
                         if serialized_shred.is_none() {
                             if let Some(slot_meta) = slot_meta {
-                                panic!(
```

Comment:
I think Carl wanted still keep the panic if we are really deterministic that has been purged. 

The idea is in purge the moment we start purging on a slot range, we mark the largest slot being purged to a atomic variable. And in case of the the read error here, we check if the slot is > that largest slot purged, if so, we still panic. otherwise return error.

Reply:
I was thinking about this approach as well but have this PR out for quick mitigation.  Let me give it a try.

Reply:
Turns out we already have such an atomic!  Originally I was searching inside blockstore_purge.rs but didn't find anything, but this is actually set in the ledger_cleanup_service.rs.